docs/dev/tutorials/custom-paymaster-tutorial.md

```diff
@@ -192,24 +225,25 @@ try
 require(success, "Failed to transfer funds to the bootloader");
 ```
 
-::: tip You should validate all the requirements first
-
-The [rules](../developer-guides/aa.md#the-validation-step) for the paymaster throttling say that the paymaster won't be throttled if the first storage read the value of which differed from the execution on the API was a storage slot that belonged to the user.
-
-That is why it is important to verify that the user provided all the allowed prerequisites to the transaction _before_ performing any logic. This is the reason we _first_ check that the user provided enough allowance, and only then do we do `transferFrom`.
+::: tip Validate all requirements first
+The [validation steps](../developer-guides/aa.md#the-validation-step) ensure that the paymaster won't throttle if the first storage read which has a different value from the execution on the API is a storage slot that belongs to the user.
 
+This is why it is important to verify transaction prerequisites _before_ performing any logic and why we _first_ check that the user provided enough allowance before calling `transferFrom`.
 :::
 
-### Full code of the paymaster
+## Create paymaster contract
```

Comment:
I find it confusing that we've gone through all the logic of the paymaster and after all that we're going to create it here.

Reply:
**Create** is an action. Going through the logic is passive learning. The instruction below is: 

Create the `contracts/MyPaymaster.sol` file and copy/paste the following:

Reply:
I'll change it a bit.